nbzip/static/tree.js

```diff
@@ -0,0 +1,62 @@
+
+
+define([
+    'jquery',
+    'base/js/utils',
+    'base/js/namespace'
+], function (
+    $, utils, Jupyter
+) {
+
+    // inspired by: https://stackoverflow.com/questions/1106377/detect-when-browser-receives-file-download
+    currToken = null;
+
+    var newToken = function () {
+      return new Date().getTime();
+    }
+
+    getCookie = function ( name ) {
```

Comment:
I think we can just use the web.authenticated decorator in the python code, and not implement all this code manually.

Reply:
oh, i'm using the cookie to check if the zipping is done to make the zip button reappear. But, adding the decorator for authentication is a good idea.